services/property-svc/internal/property-value/api/grpc.go

```diff
@@ -18,71 +23,105 @@ func NewPropertyValueService(aggregateStore hwes.AggregateStore) *PropertyValueG
 	propertyValueID := uuid.New()
 
-	subjectID, err := uuid.Parse(req.GetSubjectId())
+	propertyID, err := uuid.Parse(req.GetPropertyId())
 	if err != nil {
 		return nil, err
 	}
-	propertyID, err := uuid.Parse(req.GetPropertyId())
+
+	subjectID, err := uuid.Parse(req.GetSubjectId())
 	if err != nil {
 		return nil, err
 	}
 
 	var value interface{}
 	switch req.Value.(type) {
-	case *pb.CreatePropertyValueRequest_TextValue:
+	case *pb.AttachPropertyValueRequest_TextValue:
 		value = req.GetTextValue()
-	case *pb.CreatePropertyValueRequest_NumberValue:
+	case *pb.AttachPropertyValueRequest_NumberValue:
 		value = req.GetNumberValue()
-	case *pb.CreatePropertyValueRequest_BoolValue:
+	case *pb.AttachPropertyValueRequest_BoolValue:
 		value = req.GetBoolValue()
-	case *pb.CreatePropertyValueRequest_DateValue:
+	case *pb.AttachPropertyValueRequest_DateValue:
 		value = req.GetDateValue()
-	case *pb.CreatePropertyValueRequest_DateTimeValue:
+	case *pb.AttachPropertyValueRequest_DateTimeValue:
 		value = req.GetDateTimeValue()
+	case *pb.AttachPropertyValueRequest_SelectValue:
+		value = req.GetSelectValue()
 	default:
 		value = nil
 	}
 
-	if err := commandsV1.NewCreatePropertyValueCommandHandler(s.as)(ctx, propertyValueID, propertyID, subjectID, req.GetSubjectType(), value); err != nil {
+	if err := commandsV1.NewAttachPropertyValueCommandHandler(s.as)(ctx, propertyValueID, propertyID, value, subjectID); err != nil {
 		return nil, err
 	}
 
-	return &pb.CreatePropertyValueResponse{
-		Id: propertyValueID.String(),
+	return &pb.AttachPropertyValueResponse{
+		PropertyValueId: propertyValueID.String(),
 	}, nil
 }
 
-func (s *PropertyValueGrpcService) GetPropertyValue(ctx context.Context, req *pb.GetPropertyValueRequest) (*pb.GetPropertyValueResponse, error) {
-	id, err := uuid.Parse(req.GetId())
+func (s *PropertyValueGrpcService) GetAttachedPropertyValues(ctx context.Context, req *pb.GetAttachedPropertyValuesRequest) (*pb.GetAttachedPropertyValuesResponse, error) {
+	db := hwdb.GetDB()
+	propertyValueRepo := property_value_repo.New(db)
+	propertyRepo := property_repo.New(db)
+
+	subjectID, err := uuid.Parse(req.GetSubjectId())
 	if err != nil {
 		return nil, err
 	}
 
-	property, err := v1queries.NewGetPropertyValueByIDQueryHandler(s.as)(ctx, id)
+	propertyValues, err := v1queries.NewGetPropertyValuesBySubjectIDQueryHandler(propertyValueRepo)(ctx, subjectID)
 	if err != nil {
 		return nil, err
 	}
 
-	response := &pb.GetPropertyValueResponse{
-		Id:          property.ID.String(),
-		SubjectType: property.SubjectType,
-		PropertyId:  property.PropertyID.String(),
-	}
+	propertyValuesRes, err := hwutil.MapWithErr(propertyValues, func(propertyValue property_value_repo.PropertyValue) (*pb.GetAttachedPropertyValuesResponse_Value, error) {
+		property, err := propertyRepo.GetPropertyById(ctx, propertyValue.PropertyID)
```

Comment:
database query in for loop, big no-no! refactor into building *one* query

maybe you can even do this in a JOIN together with the previous query